platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/client/impl/SecureCxfClientFactoryImpl.java

```diff
@@ -387,11 +492,69 @@ public final T getClientForSubject(Subject subject) {
 
               T newClient = getNewClient();
 
-              if (!basicAuth && StringUtils.startsWithIgnoreCase(asciiString, HTTPS)) {
-                if (subject instanceof ddf.security.Subject) {
-                  RestSecurity.setSubjectOnClient(
-                      (ddf.security.Subject) subject, WebClient.client(newClient));
+              if (useOauth
+                  && Strings.isNotBlank(clientId)
+                  && Strings.isNotBlank(clientSecret)
+                  && Strings.isNotBlank(discoveryUrl)
+                  && oauthSecurity != null) {
+                if (CREDENTIAL_FLOW.equals(oauthFlow)) {
+                  oauthSecurity.setSystemTokenOnClient(
+                      WebClient.client(newClient), clientId, clientSecret, discoveryUrl);
+                } else {
+                  oauthSecurity.setUserTokenOnClient(
+                      WebClient.client(newClient),
+                      (ddf.security.Subject) subject,
+                      sourceId,
+                      clientId,
+                      clientSecret,
+                      discoveryUrl);
                 }
+              } else if (!basicAuth
+                  && StringUtils.startsWithIgnoreCase(asciiString, HTTPS)
+                  && subject instanceof ddf.security.Subject) {
+                RestSecurity.setSubjectOnClient(
+                    (ddf.security.Subject) subject, WebClient.client(newClient));
+              }
+
+              auditRemoteConnection(asciiString);
+
+              return newClient;
+            });
+  }
+
+  /**
+   * Clients produced by this method will be secured with two-way ssl and the provided security
+   * subject.
+   *
+   * <p>The returned client should NOT be reused between requests! This method should be called for
+   * each new request in order to ensure that the security token is up-to-date each time.
+   */
+  public final T getClientForSystemSubject(Subject subject) {
```

Comment:
`getClientForSystemSubject` is not different for SAML but if the source is using OAuth then it will attach the system's access tokens instead of the users. It's used for availability checks of the source and if the source is using Credential Flow instead of Code Flow.